### ISSUE TITLE
WiFi AP_STA does not work when stopped/started in the middle of the program (IDFGH-7095)

### ISSUE THREAD

**Issue by krupis** [espressif/esp-idf#8705]
I have configured my device in the AP_STA mode. I use wifi to scan nearby AP and also I will be later using to create my own AP webserver.

Function to initialise WIFI
```
void WIFI_INIT_AP_STA(void)
{
    ESP_ERROR_CHECK(esp_netif_init());
    ESP_ERROR_CHECK(esp_event_loop_create_default());

    ap_netif_g = esp_netif_create_default_wifi_ap();
    assert(ap_netif_g);

    sta_netif_g = esp_netif_create_default_wifi_sta();
    assert(sta_netif_g);

    wifi_init_config_t cfg = WIFI_INIT_CONFIG_DEFAULT();
    ESP_ERROR_CHECK(esp_wifi_init(&cfg));

    wifi_config_t wifi_config = {
        .ap = {
            .ssid = EXAMPLE_ESP_WIFI_SSID_AP,
            .ssid_len = strlen(EXAMPLE_ESP_WIFI_SSID_AP),
            .channel = EXAMPLE_ESP_WIFI_CHANNEL,
            .password = EXAMPLE_ESP_WIFI_PASS_AP,
            .max_connection = EXAMPLE_MAX_STA_CONN,
            .authmode = WIFI_AUTH_WPA_WPA2_PSK
        },
    };
    if (strlen(EXAMPLE_ESP_WIFI_PASS_AP) == 0) {
        wifi_config.ap.authmode = WIFI_AUTH_OPEN;
    }
    

    ESP_ERROR_CHECK(esp_wifi_set_mode(WIFI_MODE_APSTA));
    ESP_ERROR_CHECK(esp_wifi_set_config(WIFI_IF_AP, &wifi_config));
    ESP_ERROR_CHECK(esp_wifi_start());

    ESP_LOGI("WIFI AP", "wifi_init_softap finished. SSID:%s password:%s channel:%d",
             EXAMPLE_ESP_WIFI_SSID_AP, EXAMPLE_ESP_WIFI_PASS_AP, EXAMPLE_ESP_WIFI_CHANNEL);
}
```

In order to stop the WiFi I do everything in the reversed order as when starting the WiFi
Function to stop WIFI:
```
void STOP_WIFI_AP_STA(){
        esp_wifi_disconnect();            // break connection to AP
        esp_wifi_stop();                 // shut down the wifi radio
        esp_wifi_deinit();              // release wifi resources   
        esp_netif_destroy_default_wifi(ap_netif_g);
        esp_netif_destroy_default_wifi(sta_netif_g);
        esp_event_loop_delete_default();
        esp_netif_deinit();
}
```

Normally, when the device is powered up, the WIFI_INIT_AP_STA will be called for the first time, The console log:
```
I (2880) wifi:wifi driver task: 3ffcd548, prio:23, stack:6656, core=0
I (2900) wifi:wifi firmware version: 7679c42
I (2900) wifi:wifi certification version: v7.0
I (2900) wifi:config NVS flash: enabled
I (2900) wifi:config nano formating: disabled
I (2900) wifi:Init data frame dynamic rx buffer num: 32
I (2910) wifi:Init management frame dynamic rx buffer num: 32
I (2910) wifi:Init management short buffer num: 32
I (2920) wifi:Init dynamic tx buffer num: 32
I (2920) wifi:Init static rx buffer size: 1600
I (2930) wifi:Init static rx buffer num: 10
I (2930) wifi:Init dynamic rx buffer num: 32
[1B][0;32mI (2930) wifi_init: rx ba win: 6[1B][0m
[1B][0;32mI (2940) wifi_init: tcpip mbox: 32[1B][0m
[1B][0;32mI (2940) wifi_init: udp mbox: 6[1B][0m
[1B][0;32mI (2940) wifi_init: tcp mbox: 6[1B][0m
[1B][0;32mI (2950) wifi_init: tcp tx win: 5744[1B][0m
[1B][0;32mI (2950) wifi_init: tcp rx win: 5744[1B][0m
[1B][0;32mI (2960) wifi_init: tcp mss: 1440[1B][0m
[1B][0;32mI (2960) wifi_init: WiFi IRAM OP enabled[1B][0m
[1B][0;32mI (2970) wifi_init: WiFi RX IRAM OP enabled[1B][0m
[1B][0;32mI (2980) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07[1B][0m
I (3080) wifi:mode : sta (8c:4b:14:0f:2a:a0) + softAP (8c:4b:14:0f:2a:a1)
I (3080) wifi:enable tsf
I (3080) wifi:Total power save buffer number: 16
I (3080) wifi:Init max length of beacon: 752/752
I (3080) wifi:Init max length of beacon: 752/752
[1B][0;32mI (3090) WIFI AP: wifi_init_softap finished. SSID:default_ssid password:test12345 channel:1[1B][0m
```

I can now sucesfully connect to the AP:

```
I (12830) wifi:new:<1,1>, old:<1,1>, ap:<1,1>, sta:<0,0>, prof:1
I (12830) wifi:station: cc:6b:1e:9b:36:43 join, AID=1, bgn, 40U
[1B][0;32mI (12890) esp_netif_lwip: DHCP server assigned IP to a station, IP is: 192.168.4.2[1B][0m
W (16520) wifi:<ba-add>idx:2 (ifx:1, cc:6b:1e:9b:36:43), tid:0, ssn:57, winSize:64
```

I can Stop the WiFi at anytime by calling the STOP_WIFI_AP_STA function.
The console logs when I stop the WIFI:
```
I (32753) wifi:flush txq
I (32753) wifi:stop sw txq
I (32753) wifi:lmac stop hw txq
I (32753) wifi:Deinit lldesc rx mblock:10
```

Then, after sometime ( it can be days or hours later) I will try to start the WIFI again by calling WIFI_INIT_AP_STA. 
The console logs when WiFi is started again:
```
I (32890) wifi:wifi driver task: 3ffd2294, prio:23, stack:6656, core=0
I (32910) wifi:wifi firmware version: 7679c42
I (32910) wifi:wifi certification version: v7.0
I (32910) wifi:config NVS flash: enabled
I (32910) wifi:config nano formating: disabled
I (32910) wifi:Init data frame dynamic rx buffer num: 32
I (32920) wifi:Init management frame dynamic rx buffer num: 32
I (32920) wifi:Init management short buffer num: 32
I (32930) wifi:Init dynamic tx buffer num: 32
I (32930) wifi:Init static rx buffer size: 1600
I (32940) wifi:Init static rx buffer num: 10
I (32940) wifi:Init dynamic rx buffer num: 32
[1B][0;32mI (32940) wifi_init: rx ba win: 6[1B][0m
[1B][0;32mI (32950) wifi_init: tcpip mbox: 32[1B][0m
[1B][0;32mI (32950) wifi_init: udp mbox: 6[1B][0m
[1B][0;32mI (32960) wifi_init: tcp mbox: 6[1B][0m
[1B][0;32mI (32960) wifi_init: tcp tx win: 5744[1B][0m
[1B][0;32mI (32960) wifi_init: tcp rx win: 5744[1B][0m
[1B][0;32mI (32970) wifi_init: tcp mss: 1440[1B][0m
[1B][0;32mI (32970) wifi_init: WiFi IRAM OP enabled[1B][0m
[1B][0;32mI (32980) wifi_init: WiFi RX IRAM OP enabled[1B][0m
I (32990) wifi:mode : sta (8c:4b:14:0f:2a:a0) + softAP (8c:4b:14:0f:2a:a1)
I (32990) wifi:enable tsf
I (32990) wifi:Total power save buffer number: 16
[1B][0;32mI (33000) WIFI AP: wifi_init_softap finished. SSID:default_ssid password:test12345 channel:1[1B][0m
```

I can immediately see some differences between the first time WiFi was started and when it was stopped and started again. Looking at the console logs, the second time WiFi was started is missing a few logs:
```
[1B][0;32mI (2980) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07[1B][0m
I (3080) wifi:Init max length of beacon: 752/752
I (3080) wifi:Init max length of beacon: 752/752
```

After disabling and enabling the WiFi again, the WiFi is not working properly. It does not seem to give an IP to the connected station. I can see the AP appear on WiFi list but when I connect to the device, the following legs appear:
```
I (92700) wifi:new:<1,1>, old:<1,1>, ap:<1,1>, sta:<0,0>, prof:1
I (92700) wifi:station: cc:6b:1e:9b:36:43 join, AID=1, bgn, 40U
W (96090) wifi:<ba-add>idx:2 (ifx:1, cc:6b:1e:9b:36:43), tid:0, ssn:34, winSize:64
```

Notice the missing log
```
[1B][0;32mI (12890) esp_netif_lwip: DHCP server assigned IP to a station, IP is: 192.168.4.2[1B][0m
```

Could someone help me understand what is the issue and why the device does not assign IP to the station when I stop and start the WiFi ? I assume I am not Stoppping the WiFi correctly

**Expected Behavior**
Should be able to Disable the WiFi in the middle of the program and Enable anytime I want

**Actual Behavior**
Only works for the initial time. After disabling and enabling once the station no longer receives an IP address

**Comment by nishanth-radja** [comment]
@krupis can you pls mention the IDF version you are using?

**Comment by krupis** [comment]
Hey. I am using v4.4

I have been further thinking about it.. Could it be because the DHCP lease time? If that is the case, I dont think there is any way around this? I hope I am wrong about this

**Comment by krupis** [comment]
I have been testing it in both AP_STA and only AP mode. It seems to be working in AP mode but when I use AP_STA mode, It does not.

**Comment by krupis** [comment]
I have created a little repository where this issue can be easily recreated:
https://github.com/krupis/wifi_test/blob/master/main/blink_example_main.c

You can enable/disable WiFi using 2 UART commands:
"stop_wifi_ap"
and
"start_wifi_ap"

**Comment by nishanth-radja** [comment]
@krupis The issue seems to be fixed on latest 4.4, can you pull the latest 4.4 IDF and try. "4222e93351bd387b87676b6a43ccf44397958fb5"

Note:
The issue was fixed on this commit "ef7ce17dbd4a0585d4c10549d1788f40f9aa02d5"
